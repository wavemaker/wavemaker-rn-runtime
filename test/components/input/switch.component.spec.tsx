import React, { ReactNode } from 'react';
import { Text, TouchableOpacity } from 'react-native';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import WmSwitch from '@wavemaker/app-rn-runtime/components/input/switch/switch.component';

describe('Test Switch component', () => {
    test('Check validity of sample component', () => {
      const tree = renderer.create(<WmSwitch name="test_Switch"/>).toJSON();
      expect(tree).toMatchSnapshot();
    });
});