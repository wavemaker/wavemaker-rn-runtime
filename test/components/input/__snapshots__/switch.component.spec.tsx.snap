// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WmSwitch renders correctly with default props 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "minHeight": 24,
      "paddingBottom": 4,
      "paddingLeft": 4,
      "paddingRight": 4,
      "paddingTop": 4,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": false,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "overflow": "hidden",
        },
        [
          {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderBottomLeftRadius": 0,
            "borderBottomRightRadius": 0,
            "borderColor": "#cccccc",
            "borderRightWidth": 0,
            "borderTopLeftRadius": 0,
            "borderTopRightRadius": 0,
            "borderWidth": 1,
            "color": "#151420",
            "fontFamily": "Roboto",
            "fontSize": 14,
            "fontWeight": "500",
            "height": 40,
            "justifyContent": "center",
            "minWidth": 64,
            "paddingLeft": 16,
            "paddingRight": 16,
          },
          {
            "borderBottomLeftRadius": 18,
            "borderBottomRightRadius": 0,
            "borderTopLeftRadius": 18,
            "borderTopRightRadius": 0,
          },
          null,
        ],
      ]
    }
  >
    <View>
      <Text
        style={
          [
            {
              "fontFamily": "Roboto",
              "fontSize": 16,
              "fontWeight": "500",
              "textTransform": "uppercase",
              "userSelect": "none",
            },
            {
              "color": "#151420",
            },
          ]
        }
      >
        yes
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": false,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "overflow": "hidden",
        },
        [
          {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderBottomLeftRadius": 0,
            "borderBottomRightRadius": 0,
            "borderColor": "#cccccc",
            "borderRightWidth": 0,
            "borderTopLeftRadius": 0,
            "borderTopRightRadius": 0,
            "borderWidth": 1,
            "color": "#151420",
            "fontFamily": "Roboto",
            "fontSize": 14,
            "fontWeight": "500",
            "height": 40,
            "justifyContent": "center",
            "minWidth": 64,
            "paddingLeft": 16,
            "paddingRight": 16,
          },
          {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderBottomLeftRadius": 0,
            "borderBottomRightRadius": 0,
            "borderColor": "#cccccc",
            "borderRightWidth": 0,
            "borderTopLeftRadius": 0,
            "borderTopRightRadius": 0,
            "borderWidth": 1,
            "color": "#151420",
            "fontFamily": "Roboto",
            "fontSize": 14,
            "fontWeight": "500",
            "height": 40,
            "justifyContent": "center",
            "minWidth": 64,
            "paddingLeft": 16,
            "paddingRight": 16,
          },
          null,
        ],
      ]
    }
  >
    <View>
      <Text
        style={
          [
            {
              "fontFamily": "Roboto",
              "fontSize": 16,
              "fontWeight": "500",
              "textTransform": "uppercase",
              "userSelect": "none",
            },
            {
              "color": "#151420",
            },
          ]
        }
      >
        no
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": false,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "overflow": "hidden",
        },
        [
          {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderBottomLeftRadius": 0,
            "borderBottomRightRadius": 0,
            "borderColor": "#cccccc",
            "borderRightWidth": 0,
            "borderTopLeftRadius": 0,
            "borderTopRightRadius": 0,
            "borderWidth": 1,
            "color": "#151420",
            "fontFamily": "Roboto",
            "fontSize": 14,
            "fontWeight": "500",
            "height": 40,
            "justifyContent": "center",
            "minWidth": 64,
            "paddingLeft": 16,
            "paddingRight": 16,
          },
          {
            "borderBottomLeftRadius": 0,
            "borderBottomRightRadius": 18,
            "borderRightWidth": 1,
            "borderTopLeftRadius": 0,
            "borderTopRightRadius": 18,
          },
          null,
        ],
      ]
    }
  >
    <View>
      <Text
        style={
          [
            {
              "fontFamily": "Roboto",
              "fontSize": 16,
              "fontWeight": "500",
              "textTransform": "uppercase",
              "userSelect": "none",
            },
            {
              "color": "#151420",
            },
          ]
        }
      >
        maybe
      </Text>
    </View>
  </View>
</View>
`;
